website/docs/reference/resource-configs/deltastream-configs.md

```diff
@@ -0,0 +1,738 @@
+| `topic`        | The topic name for the stream output.                                                         | Required  |
+| `value.format` | Format for the stream values (e.g., 'json', 'avro').                                          | Required  |
+| `key.format`   | Format for the stream keys (e.g., 'primitive', 'json').                                       | Optional  |
+| `key.type`     | Data type for the stream keys (e.g., 'VARCHAR', 'BIGINT').                                    | Optional  |
+| `timestamp`    | Column name to use as the event timestamp.                                                    | Optional  |
+
+### Changelog Materialization
+
+Captures changes in the data stream:
+
+**Project file configuration:**
+```yaml
+models:
+  <resource-path>:
+    +materialized: changelog
+    +parameters:
+      topic: 'changelog_topic'
+      value.format: 'json'
+    +primary_key: [column_name]
+```
+
+**Config block configuration:**
+```sql
+{{ config(
+    materialized='changelog',
+    parameters={
+        'topic': 'order_updates',
+        'value.format': 'json'
+    },
+    primary_key=['order_id']
+) }}
+
+SELECT 
+    order_id,
+    status,
+    updated_at
+FROM {{ ref('orders_stream') }}
+```
+
+#### Changelog Configuration Options
+
+| Option         | Description                                                                                   | Required? |
+|----------------|-----------------------------------------------------------------------------------------------|-----------|
+| `materialized` | How the model will be materialized. Must be `changelog` to create a changelog model.          | Required  |
+| `topic`        | The topic name for the changelog output.                                                      | Required  |
+| `value.format` | Format for the changelog values (e.g., 'json', 'avro').                                       | Required  |
+| `primary_key`  | List of column names that uniquely identify rows for change tracking.                         | Required  |
+
+### Materialized View
+
+Creates a continuously updated view:
+
+**Config block configuration:**
+```sql
+{{ config(materialized='materialized_view') }}
+
+SELECT 
+    product_id,
+    COUNT(*) as purchase_count
+FROM {{ ref('purchase_events') }}
+GROUP BY product_id
+```
+
+## YAML-Only Resource Configurations
+
+DeltaStream supports two types of model definitions for infrastructure components:
+
+1. **Managed Resources (Models)** - Automatically included in the dbt DAG
+2. **Unmanaged Resources (Sources)** - Created on-demand using specific macros
+
+### Should You Use Managed or Unmanaged Resources?
+
+If you plan to be able to recreate all the infrastructure in different environments and/or use graph operators to execute only the creation of specific resources and downstream transformations, you should use managed resources. Otherwise, it might be simpler to use unmanaged resources to avoid placeholder files.
```

Comment:
```suggestion
- Use managed resources if you plan to recreate all the infrastructure in different environments and/or use graph operators to execute only the creation of specific resources and downstream transformations.
- Otherwise, it might be simpler to use unmanaged resources to avoid placeholder files.
```